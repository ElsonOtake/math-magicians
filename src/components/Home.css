main {
  width: 80%;
  margin: 2rem 10%;
}

header,
h4,
p {
  text-align: left;
}

header {
  font-size: 2rem;
  color: blue;
}

h4 {
  font-size: 1rem;
}

p {
  margin: 0;
  line-height: 2rem;
  color: black;
}
