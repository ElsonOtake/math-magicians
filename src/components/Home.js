import React from 'react';

const Home = () => (
  <>
    <header>Welcome to our page</header>
    <h1>Mathemagician</h1>
    <p>From Wikipedia, the free encyclopedia</p>
    <p>A mathemagician is a mathematician who is also a magician.</p>
    <p>
      {`
      The term &quot;mathemagic&quot; is believed to have been introduced by Royal
       Vale Heath with his 1933 book &quot;Mathemagic&quot;.
      `}
    </p>
    <p>
      {`
      The name &quot;mathemagician&quot; was probably first applied to Martin Gardner, 
      but has since been used to describe many mathematician/magicians, including 
      Arthur T. Benjamin, Persi Diaconis, and Colm Mulcahy. Diaconis has 
      suggested that the reason so many mathematicians are magicians is that &quot;inventing 
      a magic trick and inventing a theorem are very similar activities.&quot;
      `}
    </p>
    <p>
      {`
      Mathemagician is a neologism, specifically a portmanteau, that combines 
      mathematician and magician. A great number of self-working mentalism tricks 
      rely on mathematical principles. Max Maven often utilizes this type of magic in his performance.
      `}
    </p>
    <p>
      {`
      The Mathemagician is the name of a character in the 1961 children's 
      book The Phantom Tollbooth. He is the ruler of Digitopolis, the kingdom of mathematics.
      `}
    </p>
  </>
);

export default Home;
