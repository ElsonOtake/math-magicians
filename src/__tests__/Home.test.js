import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import Home from '../components/Home';

describe('Tests for the Home component', () => {
  render(<Home />);
  test('Check for the following test on screen', () => {
    expect(screen.getByText('Welcome to our page')).toBeInTheDocument();
    expect(screen.getByText(/Wikipedia/)).toBeInTheDocument();
    expect(screen.getByText(/Digitopolis/)).toBeInTheDocument();
  });
  test('Check for the snapshot', () => {
    expect(screen.debug()).toMatchSnapshot();
  });
});
