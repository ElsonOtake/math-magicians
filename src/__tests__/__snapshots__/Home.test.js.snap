// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for the Home component Check for the snapshot 1`] = `undefined`;
